solutions/CSES Graph Girth.mdx

```diff
@@ -26,234 +26,70 @@ This algorithm runs in $\mathcal{O}(N + M)$ time. Since $N$ and $M$ are so small
-template<size_t SZ> str ts(bitset<SZ> b) {
-	str res = ""; F0R(i,SZ) res += char('0'+b[i]);
-	return res; }
-tcTU> str ts(pair<T,U> p);
-tcT> str ts(T v) { // containers with begin(), end()
-	#ifdef LOCAL
-		bool fst = 1; str res = "{";
-		for (const auto& x: v) {
-			if (!fst) res += ", ";
-			fst = 0; res += ts(x);
-		}
-		res += "}"; return res;
-	#else
-		bool fst = 1; str res = "";
-		for (const auto& x: v) {
-			if (!fst) res += " ";
-			fst = 0; res += ts(x);
+		for (int adj_node : adj[node]) {
+			if (dist[adj_node] == -1) {
+				dist[adj_node] = dist[node] + 1;
+				bfs.push(adj_node);
+			}
+			else {
+				if (dist[adj_node] >= dist[node]) {
+					ans = min(ans, 1 + dist[adj_node] + dist[node]);
+				}
+			}
 		}
-		return res;
-
-	#endif
-}
-tcTU> str ts(pair<T,U> p) {
-	#ifdef LOCAL
-		return "("+ts(p.f)+", "+ts(p.s)+")"; 
-	#else
-		return ts(p.f)+" "+ts(p.s);
-	#endif
-}
-
-// OUTPUT
-tcT> void pr(T x) { cout << ts(x); }
-tcTUU> void pr(const T& t, const U&... u) { 
-	pr(t); pr(u...); }
-void ps() { pr("\n"); } // print w/ spaces
-tcTUU> void ps(const T& t, const U&... u) { 
-	pr(t); if (sizeof...(u)) pr(" "); ps(u...); }
-
-// DEBUG
-void DBG() { cerr << "]" << endl; }
-tcTUU> void DBG(const T& t, const U&... u) {
-	cerr << ts(t); if (sizeof...(u)) cerr << ", ";
-	DBG(u...); }
-#ifdef LOCAL // compile with -DLOCAL, chk -> fake assert
-	#define dbg(...) cerr << "Line(" << __LINE__ << ") -> [" << #__VA_ARGS__ << "]: [", DBG(__VA_ARGS__)
-	#define chk(...) if (!(__VA_ARGS__)) cerr << "Line(" << __LINE__ << ") -> function(" \
-		 << __FUNCTION__  << ") -> CHK FAILED: (" << #__VA_ARGS__ << ")" << "\n", exit(0);
-#else
-	#define dbg(...) 0
-	#define chk(...) 0
-#endif
+	}
 
-void setPrec() { cout << fixed << setprecision(15); }
-void unsyncIO() { cin.tie(0)->sync_with_stdio(0); }
-// FILE I/O
-void setIn(str s) { freopen(s.c_str(),"r",stdin); }
-void setOut(str s) { freopen(s.c_str(),"w",stdout); }
-void setIO(str s = "") {
-	unsyncIO(); setPrec();
-	// cin.exceptions(cin.failbit); 
-	// throws exception when do smth illegal
-	// ex. try to read letter into int
-	if (sz(s)) setIn(s+".in"), setOut(s+".out"); // for USACO
+	return ans;
 }
 
-int n,m,dist[2501],ans=MOD;
-vi adj[2501];
-
 int main() {
-	setIO(); re(n,m);
-	F0R(i,m) {
-		int a,b; re(a,b);
-		adj[a].pb(b), adj[b].pb(a);
+	cin >> n >> m;
+	for (int i = 0; i < m; i++) {
+		int a, b;
+		cin >> a >> b;
+		a--; b--;
+		adj[a].push_back(b);
+		adj[b].push_back(a);
 	}
-	FOR(i,1,n+1) {
-		FOR(j,1,n+1) dist[j] = MOD;
-		queue<int> q; q.push(i); dist[i] = 0;
-		while (sz(q)) {
-			int x = q.front(); q.pop();
-			trav(t,adj[x]) {
-				if (dist[t] == MOD)
-					dist[t] = dist[x]+1, q.push(t);
-				else if (dist[t] >= dist[x])
-					ckmin(ans,dist[t]+dist[x]+1);
-			}
-		}
+
+	int res = inf;
+	for (int i = 0; i < n; i++) {
+		res = min(res, cycle_len(i));
 	}
-	ps(ans == MOD ? -1 : ans);
+
+	if (res == inf) {
+		cout << -1 << endl;
+		return 0;
+	}
+
+	cout << res << endl;
```

Comment:
the ternary expression was shorter... this is okay, up to you whether to change it or not

Reply:
Idt it matters lol...

Reply:
lol i like ternary expressions as well, but its up to u